eventrouter/internal/eventrouter_os.c

```diff
@@ -642,6 +669,8 @@ void ErUnsubscribe(ErModule_t *a_module, ErEventType_t a_event_type)
 
 ErEvent_t *ErReceive(void)
 {
+    WaitUntilInitComplete();
```

Comment:
Hmm, it seems a little wasteful to call this on every `*Receive`.  The problem is tasks themselves, in which case you can, say, require ER tasks to wait on a semaphore as a first step, before proceeding to their normal business.
When ER is done creating all tasks/queues, it can post to the semaphore (as many times as needed to unblock all tasks) and the rest can proceed normally. This can also simplify the locking logic.

Reply:
From the performance point of view it is minimal, yes.
From the implementation and understanding point of view, it is strange to see `WaitUntilInitComplete()` be called in every `*Receive` call. Again, if the problem is that the tasks run before they are told to do so, it is IMO better to solve that problem, than to spill it into other abstractions that should just work the same.

Reply:
Is it reasonable to say that it's a problem for POSIX tasks to do other work before the event router is initialized? The only thing we must prevent is interaction with the message-passing primitives.

Would it help to call the function `EnsureInitComplete()`?

Reply:
> Is it reasonable to say that it's a problem for POSIX tasks to do other work before the event router is initialized?
I think this is reasonable. We want all usages (FreeRTOS and POSIX) to follow the same model: init everything, then start running the tasks.

Again, I think it's OK to submit this as is. We can make changes in the future if necessary.